monitor-manage-device-with-iotcentral/CoffeeMaker.js

```diff
@@ -0,0 +1,266 @@
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+var ProvisioningTransport = require('azure-iot-provisioning-device-mqtt').Mqtt;
+var SymmetricKeySecurityClient = require('azure-iot-security-symmetric-key').SymmetricKeySecurityClient;
+var ProvisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
+
+// DPS connection information
+var modelIdObject = { modelId: 'dtmi:com:example:ConnectedCoffeeMaker;1' };
+var provisioningHost = 'global.azure-devices-provisioning.net';
+var idScope = '{ID scope}';
+var registrationId = '{Device ID}';
+var symmetricKey = '{Primary key}';
+var provisioningSecurityClient = new SymmetricKeySecurityClient(registrationId, symmetricKey);
+var provisioningClient = ProvisioningDeviceClient.create(provisioningHost, idScope, new ProvisioningTransport(), provisioningSecurityClient);
+var centralClient;
+
+// Global variables
+var optimalTemperature = 96;
+var cupState = 'detected';
+var brewingState = 'notbrewing';
+var cupTimer = 20;
+var brewingTimer = 0;
+var maintenanceState = false;
+var warrantyState = Math.random() < 0.5 ? true : false;
+
+// Helper function to produce nice numbers (##.#)
+function niceNumber(value) {
+    var number = (Math.round(value * 10.0)).toString();
+    return number.substr(0, 2) + '.' + number.substr(2, 1);
+}
+
+// <Telemetry>
+// Send device simulated telemetry measurements
+function sendTelemetry() {
+    // Simulate the telemetry values
+    var temperature = optimalTemperature + (Math.random() * 4) - 2;
+    var humidity = 20 + (Math.random() * 80);
+
+    // Cup timer - every 20 seconds randomly decide if the cup is present or not
+    cupTimer--;
+
+    if (cupTimer == 0) {
+        cupTimer = 20;
+        cupState = Math.random() < 0.5 ? 'detected' : 'notdetected';
+    }
+
+    // Brewing timer
+    if (brewingTimer > 0) {
+        brewingTimer--;
+
+        // Finished brewing
+        if (brewingTimer == 0) {
+            brewingState = 'notbrewing';
+        }
+    }
+
+    // Create the data JSON package
+    var data = JSON.stringify(
+        {
+            WaterTemperature: temperature,
+            AirHumidity: humidity,
+            CupDetected: cupState,
+            Brewing: brewingState
+        });
+
+    // Create the message with the above defined data
+    var message = new Message(data);
+
+    // Show the information in console
+    var infoTemperature = niceNumber(temperature);
+    var infoHumidity = niceNumber(humidity);
+    var infoCup = (cupState == 'detected') ? 'Y' : 'N';
+    var infoBrewing = (brewingState == 'brewing') ? 'Y' : 'N';
+    var infoMaintenance = maintenanceState ? 'Y' : 'N';
+
+    console.log('Telemetry send: Temperature: ' + infoTemperature +
+        ' Humidity: ' + infoHumidity + '%' +
+        ' Cup Detected: ' + infoCup +
+        ' Brewing: ' + infoBrewing +
+        ' Maintenance Mode: ' + infoMaintenance);
```

Comment:
style nit: [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are easier to read than using the `+` operator to concatenate strings